### PR TITLE
Added event horizon and ergosphere for Kerr black-hole

### LINKED ISSUE #114: Implement the event horizon and ergosphere of Kerr black holes
🐞 **What bugs you?**
Event horizon is the point of no return in a black hole from which object can escape only with a speed greater than that of light. Ergosphere is an region around black hole where objects are forced to co-rotate with the black hole due to frame dragging. In kerr black holes:
  - Formula for event horizon : R<sub>H</sub> = 0.5(Rs + sqrt(Rs<sup>2</sup> - 4*a<sup>2</sup>))
    - See https://en.wikipedia.org/wiki/Kerr_metric#Important_surfaces
  - Formula for ergosphere : 0.5(Rs + sqrt(Rs<sup>2</sup> - 4*a<sup>2</sup>*cos(theta)<sup>2</sup>))
    - See https://en.wikipedia.org/wiki/Kerr_metric#Important_surfaces

We can plot the erogosphere after the implementation.

🎯 **Goal**
  - Implement `event_horizon()` in `utils/kerr_utils.py` as a function of `r, Rs, a` and give option to return in radius in BL(Boyer-Lindquist) as well as Spherical Coordinates. BL->spherical can be implemented by doing BL->cartesian->spherical
  - Implement `ergosphere()` in `utils/kerr_utils.py` as a function of `r, theta, Rs, a` and give option to return radius in BL & spherical coordinates where Rs is schwarzschild radius.

Coordinate conversion utils are already implemented
The above formulas returns values in BL coordinates

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #114`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by ritzvik**
Fixes #114 

  - Increased number of permissible lines in codeclimate. Maybe, it was inevitable.
  - made a separate constant for pi/2. Computing `np.pi/2` each time can be expensive as it is extensively called in the library. Eventually all `np.pi/2` in the library would be replaced.
  - Finally, added functions to calculate radius of event horizon & ergosphere in both Spherical & Boyer-Lindquist coordinates.

### CODE REVIEW

**Comment by ritzvik** [comment]
Also attempted to improve quality in accordance with code-factor suggestions

**Comment by shreyasbapat** [comment]
Great! I will review soon!

**Comment by ritzvik** [comment]
@shreyasbapat I think this is ready for merging!

File: src/einsteinpy/constant.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
 from astropy import constants
 
 c = constants.c.value * (u.m / u.s)
-
 G = constants.G.value * ((u.m ** 3) / (u.kg * u.s ** 2))
+pi_by_2 = 1.5707963267948965579989817342720925807952880859375
```
Comment (shreyasbapat) [inline src/einsteinpy/constant.py]: Please make this a astropy constant

File: src/einsteinpy/ijit.py [hunk 2]
```diff
@@ -17,9 +17,10 @@ def _jit(f):
         return f
 
     if inspect.isfunction(first):
-        return first
+        retval = first
     else:
-        return _jit
+        retval = _jit
+    return retval
```
Comment (shreyasbapat) [inline src/einsteinpy/ijit.py]: Why these extra lines of code?

File: src/einsteinpy/tests/test_utils/test_kerr_utils.py [hunk 3]
```diff
@@ -1,11 +1,62 @@
...
+    for t in tmp:
+        i = int(t / (4 ** 2)) % 4
+        k = int(t / 4) % 4
+        l = t % 4
+        for m in range(4):
+            chl2[i, k, l] += invg[i, m] * (
+                dmdx[l, m, k] + dmdx[k, m, l] - dmdx[m, k, l]
+            )
+    chl2 = np.multiply(chl2, 0.5)
+    # compare
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_utils/test_kerr_utils.py]: Please remove this comment

File: src/einsteinpy/utils/kerr_utils.py [hunk 4]
```diff
@@ -331,3 +334,87 @@ def nonzero_christoffels():
...
+    return J / (M * c)
+
+
+def event_horizon(Rs, a, theta=constant.pi_by_2, coord="BL"):
+    """
+    Calculate the radius of event horizon of Kerr black hole
+
+    Parameters
+    ----------
+    Rs : float
```
Comment (shreyasbapat) [inline src/einsteinpy/utils/kerr_utils.py]: Better variable name for schwarzschild radiius?

File: src/einsteinpy/utils/kerr_utils.py [hunk 5]
```diff
@@ -331,3 +334,87 @@ def nonzero_christoffels():
...
+    if coord == "BL":
+        ans = np.array([Rh, theta], dtype=float)
+    else:
+        ans = utils.CartesianToSpherical_pos(
+            utils.BLToCartesian_pos(np.array([Rh, theta, 0.0]), a)
+        )[:2]
+    return ans
+
+
+def ergosphere(Rs, a, theta=constant.pi_by_2, coord="BL"):
```
Comment (shreyasbapat) [inline src/einsteinpy/utils/kerr_utils.py]: Rename it to radius_ergosphere for more clarity!

File: docs/source/jupyter.rst [hunk 6]
```diff
@@ -7,3 +7,4 @@ Jupyter notebooks
    /examples/Symbolically Understanding Christoffel Symbol and Riemann Curvature Tensor using EinsteinPy.ipynb
    /examples/Analysing Earth using EinsteinPy!.ipynb
    /examples/Visualizing frame dragging in Kerr spacetime.ipynb
+   /examples/Visualizing event horizon and ergosphere of Kerr black hole.ipynb
```
Comment (shreyasbapat) [inline docs/source/jupyter.rst]: Add the notebook's best plot/image as `kerrblackhole.png` in the same examples folder.

File: src/einsteinpy/constant.py (hunk 1 above)
Comment (shreyasbapat) [reply src/einsteinpy/constant.py]: Also, I don't believe its a good name

Comment (ritzvik) [reply src/einsteinpy/constant.py]: I'll be removing it alltogether

File: src/einsteinpy/ijit.py (hunk 2 above)
Comment (ritzvik) [reply src/einsteinpy/ijit.py]: It fixes one of the codefactor issues

File: src/einsteinpy/utils/kerr_utils.py (hunk 4 above)
Comment (ritzvik) [reply src/einsteinpy/utils/kerr_utils.py]: R<sub>s</sub> is used everywhere in literature. Plus, it's been used in all other functions of `kerr_utils.py`

File: src/einsteinpy/utils/kerr_utils.py (hunk 5 above)
Comment (ritzvik) [reply src/einsteinpy/utils/kerr_utils.py]: fair enough!

File: docs/source/jupyter.rst (hunk 6 above)
Comment (ritzvik) [reply docs/source/jupyter.rst]: I'll do it

File: src/einsteinpy/ijit.py (hunk 2 above)
Comment (ritzvik) [reply src/einsteinpy/ijit.py]: Still, I'll do it

**Comment by ritzvik** [comment]
Did the changes as required. Did not change the `Rs` as parameter name, as it is heavily used in papers as far as I have read. @shreyasbapat see to it.

**Comment by shreyasbapat** [comment]
I believe the changes are mergeable!

**Pull request merged**